### DISCUSSION TITLE
Welcome to build123d Discussions!

### DISCUSSION THREAD

**Discussion by gumyr** [gumyr/build123d#160] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by chantran22** [comment]
Hi, I tried to install build123d but it does not work, I dont know why.
It runs alone like cq_editor or need to call in freecad or else?

Can help to clarify because i am quite new about build123d. Thanks

**Reply by jdegenstein** [reply]
I recommend you install build123d via the OCP CAD Viewer extension in VSCode. See instructions here: https://github.com/bernhard-42/vscode-ocp-cad-viewer

**Comment by smoebody** [comment]
Hi,

my name is Ulf and I was made aware of this project while searching for a replacement for OpenSCAD, which unfortunately does not integrate well in my IDE (lack of a proper preview) and other 3D-approach shortcomings.

I use a vscode-like IDE (openvscode-server) and are quite happy with my setup right now (works in browser, YAY!). I use it for designing my 3D printing parts.

Build123d has a very steep learning curve and there are many concept that I have a hard time wrapping my brain around it. If this is the way to share help and hints with the community I hope to get some best practices and advice along my journey with Build123d.

I on the other side try to help where i can (I'm not much of a python programmer right now). I certainly appreciate the work this project has seen. 

Thanks for that already.

cheers
Ulf

**Reply by gumyr** [reply]
Hi Ulf, welcome to the build123d community.  

On the front page of the [build123d documentation](https://build123d.readthedocs.io/en/latest/index.html) you will see a link to the Discord server which host the build123d community which is very active with many users and contributors that actively help each other.

As build123d is a Python library and not a traditional CAD package with an integrated viewer you'll need a separate viewer - the recommended viewer is [vscode-ocp-cad-viewer](https://github.com/bernhard-42/vscode-ocp-cad-viewer) which can either operate from within vscode or as an independent browser based viewer with your editor of choice.  

To get started I recommend you work through the [introductory examples](https://build123d.readthedocs.io/en/latest/introductory_examples.html) which start at a very basic level and work through much of the functionality of build123d. As you're coming from OpenSCAD I also recommend you read through the section of the docs on [Transitioning from OpenSCAD](https://build123d.readthedocs.io/en/latest/OpenSCAD.html#).

Quite a bit of effort has been put into the documentation to help users get up to speed.  There are pdf and even epub versions (links off the main documentation page)  if you would like to read while you're away from your computer.

One of the cool things about the Discord group is the #showcase channel where users post their creations. It's always fascinating what people create. Keep in mind that the objects you create can be exported as industry standard STEP files which are supported by manufacturing processes like CNC machining, laser cutting, etc. which opens up many more possibilities than 3d printing which one gets from STL files generated by tools like OpenSCAD.

Cheers,
Roger
